front/app/components/UI/SearchInput/index.tsx

```diff
@@ -16,63 +17,62 @@ export interface Props {
   onChange: (arg: string | null) => void;
   className?: string;
   size?: SearchInputProps['size'];
+  // This prop will ensure that screen readers
+  // get notified when the number of results have changed.
+  a11y_numberOfSearchResults: number;
 }
 
-const SearchInputWrapper = memo<Props & InjectedIntlProps>(
-  ({
-    placeholder,
-    ariaLabel,
-    debounce,
-    setClearButtonRef,
-    onChange,
-    className,
-    size,
-    intl: { formatMessage },
-  }) => {
-    const [searchTerm, setSearchTerm] = useState<string | null>(null);
+const SearchInputWrapper = ({
+  placeholder,
+  ariaLabel,
+  debounce,
+  setClearButtonRef,
+  onChange,
+  className,
+  size,
+  intl: { formatMessage },
+  a11y_numberOfSearchResults,
+}: Props & InjectedIntlProps) => {
+  const [searchTerm, setSearchTerm] = useState<string | null>(null);
 
-    const handleOnChange = useCallback(
-      (searchTerm: string | null) => {
-        setSearchTerm(searchTerm);
-        onChange(searchTerm);
-      },
-      [onChange]
-    );
+  const handleOnChange = (searchTerm: string | null) => {
+    setSearchTerm(searchTerm);
+    onChange(searchTerm);
+  };
 
-    const handleClearButtonRef = useCallback(
-      (element: HTMLButtonElement) => {
-        setClearButtonRef?.(element);
-      },
-      [setClearButtonRef]
-    );
+  const handleClearButtonRef = (element: HTMLButtonElement) => {
+    setClearButtonRef?.(element);
+  };
 
-    return (
-      <>
-        <Label htmlFor="search-input" hidden>
-          {formatMessage(messages.searchLabel)}
-        </Label>
-        <SearchInput
-          id="search-input"
-          placeholder={placeholder || formatMessage(messages.searchPlaceholder)}
-          ariaLabel={ariaLabel || formatMessage(messages.searchAriaLabel)}
-          debounce={debounce}
-          className={className}
-          setClearButtonRef={handleClearButtonRef}
-          onChange={handleOnChange}
-          i18nRemoveSearchTermMessage={formatMessage(messages.removeSearchTerm)}
-          i18nSearchTermMessage={formatMessage(messages.a11y_searchTerm, {
-            searchTerm,
-          })}
-          i18nSearchTermBlankMessage={formatMessage(
-            messages.a11y_searchTermBlank
-          )}
-          size={size}
-        />
-      </>
-    );
-  }
-);
+  return (
+    <>
+      <ScreenReaderOnly aria-live="assertive">
+        {formatMessage(messages.a11y_searchResultsHaveChanged, {
+          numberOfSearchResults: a11y_numberOfSearchResults,
+        })}
+      </ScreenReaderOnly>
```

Comment:
Moved this inside the search input so devs don't need to think about defining this outside/next to their search input each time.